src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java

```diff
@@ -120,12 +129,36 @@ private void configChange() {
       Config config = configBuilder.build();
       openShiftClient = new DefaultOpenShiftClient(config);
 
+      final BuildConfigWatcher watcher = new BuildConfigWatcher(namespace);
+      final BuildConfigList buildConfigs;
       if (namespace != null && !namespace.isEmpty()) {
-        watch = openShiftClient.buildConfigs().inNamespace(namespace).watch(new BuildConfigWatcher());
+        watch = openShiftClient.buildConfigs().inNamespace(namespace).watch(watcher);
+        buildConfigs = openShiftClient.buildConfigs().inNamespace(namespace).list();
```

Comment:
It'd be nice if the kubernetes client could deal with that automatically - a listAndWatch() method that automatically filtered out old or duplicates